master/buildbot/scripts/populatedatabase.py

```diff
@@ -0,0 +1,206 @@
+    :param projects: a list of a ProjectConfig objects
+    :param user_names: a list of an usernames (identifier) from the database
+    """
+    if verbose:
+        print("Starting creating builds")
+    created = 0
+    completed_results = [
+        results.SUCCESS,
+        results.WARNINGS,
+        results.FAILURE,
+        results.SKIPPED,
+        results.EXCEPTION,
+        results.CANCELED,
+        results.NOT_REBUILT,
+        results.DEPENDENCY_FAILURE,
+        results.MERGED,
+        results.INTERRUPTED,
+    ]
+
+    for number in range(build_count):
+        builder = random.choice(builders_list)
+        codebases = random.choice(projects[builder.project].codebases)
+        codebase = random.choice(codebases.keys())
+        repository = codebases[codebase]
+        submitted_at = datetime2epoch(
+            datetime.datetime.now() + datetime.timedelta(seconds=random.randint(-3*60*60, -3*60*60))
+        )
+        complete_at = submitted_at + random.randint(60 * 60, 3 * 60 * 60)
+        build = {
+            'branch': repository['branch'],
+            'revision': "%032x" % random.getrandbits(160),  # Random sha-1 hash
+            'repository': repository['repository'],
+            'codebase': codebase,
+            'project': builder.project,
+            'reason': 'A build was forced by {username} {username}@localhost'.format(username=random.choice(user_names)),
+            'submitted_at': submitted_at,
+            'complete_at': complete_at,
+            'buildername': builder.name,
+            'slavepool': None,
+            'number': number,
+            'slavename': random.choice(builder.slavenames),
+            'results': random.choice(completed_results),
+        }
+        result = yield db.builds.createFullBuildObject(**build)
+        if result:
+            created += 1
+
+        if verbose:
+            print_progress_bar(number+1, build_count)
+
+    if verbose:
+        print("Created %d new builds, %d skipped" % (created, build_count - created))
+
+
+def print_progress_bar(iteration, total, prefix='', suffix='', decimals=1, length=100, fill=u'█'):
+    """
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        length      - Optional  : character length of bar (Int)
+        fill        - Optional  : bar fill character (Str)
+    """
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+    filled_length = int(length * iteration // total)
+    bar = fill * filled_length + '-' * (length - filled_length)
+    print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
```

Comment:
good job. But Python has a package for this: https://pypi.python.org/pypi/progress_bar/8
We can add this package do requirements-dev (as Diogo suggests)

Reply:
Solved 👍